### PR TITLE
refactor: Remove last overloading of 'server' module name

### PR DESCRIPTION
**Pull request by alamb**
This continues @carols10cents  work on https://github.com/influxdata/influxdb_iox/pull/627 to remove the number of things called `server` IOx, this renames the server module in influxdb_iox to influxdb_ioxd

Before this PR:
```
alamb@ip-10-0-0-49 influxdb_iox % find . -name 'server' | grep -v target
./server
./src/server
```

After this PR:
```
alamb@ip-10-0-0-49 influxdb_iox % find . -name 'server' | grep -v target
./server
```

### CODE REVIEW

**Review by carols10cents — APPROVED** [review]
Thank you!!!! Love it!!

**Review by domodwyer — APPROVED** [review]

File: src/influxdb_ioxd/names.rs [hunk 1]
```diff
@@ -1,6 +1,3 @@
-pub mod http_routes;
```
Comment (pauldix) [inline src/influxdb_ioxd/names.rs:1]: Might it make more sense to just move this into the `data_types` crate?

**Review by pauldix — APPROVED** [review]
Looks great, just one small comment about maybe moving org and bucket to the data types crate. `names` also feels like a weird nondescript module name to me. Maybe something about conversion/compatibility?

File: src/influxdb_ioxd/names.rs (hunk 1 above)
Comment (alamb) [reply src/influxdb_ioxd/names.rs:1]: Yes, I think that would make sense. I'll do so as a follow on PR if you don't mind.

Comment (alamb) [reply src/influxdb_ioxd/names.rs:1]: Actually, I need to rebase anyways so I'll do it as part of this PR

**Pull request merged**